astroplan/core.py

```diff
@@ -1114,10 +1115,65 @@ def moon_illumination(self, time):
             the `~astropy.time.Time` initializer, so it can be anything that
             `~astropy.time.Time` will accept (including a `~astropy.time.Time`
             object).
+
+        moon : `~astropy.coordinates.SkyCoord` or `None` (default)
```

Comment:
It was meant to be a performance thing, since you're likely to ask for the illumination and position of the moon at the same time. I'll remove that feature for now and hold onto caching for discussion later – there are bunch of other places where we should cache too.